### ISSUE TITLE
MD001 fails to detect <hN> tag

### ISSUE THREAD

**Issue by xuhdev** [DavidAnson/markdownlint-cli2#442]
markdownlint-cli2 reports error for the following markdown file:

```md
# H1

<h2>H2</h2>

### H3

```

```
a.md:5 MD001/heading-increment Heading levels should only increment by one level at a time [Expected: h2; Actual: h3]
```

**Comment by xuhdev** [comment]
Re-reported at https://github.com/DavidAnson/markdownlint/issues/1390
